cmd/apply.go

```diff
@@ -59,19 +62,70 @@ func runApply(namespace string) error {
 		logrus.WithFields(logrus.Fields{
 			"namespace": namespace,
 		}).Info("Waiting for namespace to be ready...")
-		//init progress bar
-		uiprogress.Start()
-		bar := uiprogress.AddBar(100).AppendCompleted().PrependElapsed()
 
-		if err := api.WaitForNamespaceReady(namespace, timeout, bar); err != nil {
+		if err := waitForNamespaceReady(api.Namespaces(), namespace); err != nil {
 			return err
 		}
-
-		logrus.WithFields(logrus.Fields{
-			"namespace": namespace,
-		}).Info("Namespace is ready")
-
 	}
 
 	return nil
 }
+
+func waitForNamespaceReady(ns resource.NamespaceService, namespace string) error {
+
+	//Starting watching namespace
+	go func() {
+		err := ns.WatchNamespaces()
+		if err != nil {
+			logrus.Errorf("Error while trying to watch namespace : %s", err.Error())
+		}
+	}()
+
+	//register for event listening
+	ns.AddListener("cli_progress_bar")
+
+	//initiate progress bar
+	uiprogress.Start()
+	bar := uiprogress.AddBar(100).AppendCompleted().PrependElapsed()
+
+	//Init timer
+	timerCh := time.NewTimer(timeout).C
+	doneCh := make(chan bool)
```

Comment:
I suggest to make a buffered chanel `make(chan bool, 1)` because you will only put one value inside and exist the func after so the goroutine will stop